js/src/components/paid-ads/asset-group/asset-group-section.js

```diff
@@ -135,7 +136,70 @@ export default function AssetGroupSection() {
 						numOfIssues={ 0 }
 						initialExpanded
 					>
-						Initially expanded asset field
+						<TextsEditor
+							minNumberOfTexts={ 3 }
+							maxNumberOfTexts={ 5 }
+							maxCharacterCounts={ [ 15, 30, 30, 30, 30 ] }
+							placeholder={ __(
+								'Headline',
+								'google-listings-and-ads'
+							) }
+							addButtonText={ __(
+								'Add headline',
+								'google-listings-and-ads'
+							) }
+							initialTexts={ [ 'Hello', 'World' ] }
+							onChange={ ( v ) =>
+								// eslint-disable-next-line no-console
+								console.log( 'Headline:onChange', v )
+							}
+						/>
+					</AssetField>
+					<AssetField
+						heading={ __(
+							'Long headlines',
+							'google-listings-and-ads'
+						) }
+						subheading={ __(
+							'At least 1 required. Add up to 5.',
+							'google-listings-and-ads'
+						) }
+						help={
+							<>
+								<p>
+									{ __(
+										'The long headline is the first line of your ad, and appears instead of your short headline in larger ads. Long headlines can be up to 90 characters, and may appear with or without your description.',
+										'google-listings-and-ads'
+									) }
+								</p>
+								<p>
+									{ __(
+										'The length of the rendered headline will depend on the site it appears on. If shortened, it will end with an ellipsis(…).',
+										'google-listings-and-ads'
+									) }
+								</p>
+							</>
+						}
+						numOfIssues={ 0 }
+					>
+						<TextsEditor
+							minNumberOfTexts={ 1 }
+							maxNumberOfTexts={ 5 }
+							maxCharacterCounts={ 90 }
```

Comment:
❓  What is the rational behind supporting single number as a type for `maxCharacterCounts`?  I don't have by hand the API response format. But I think it would be more consistent to stick into array format as we do in `initialTexts`

Reply:
Based on [Google's asset specs](https://developers.google.com/google-ads/api/docs/performance-max/assets), the headlines and descriptions each have two different character limits, which are also reflected in the UI design. The single number is used for asset texts with the same character limit.
<img width="713" alt="2023-01-19 14 51 01" src="https://user-images.githubusercontent.com/17420811/213375113-022be651-5588-4af4-944b-3825a55d2835.png">

Reply:
So does that mean using `maxCharacterCounts={ 9 } `is a way to say "ok, apply 9 to all the values" whereas  `maxCharacterCounts={ [9,8,7] }` applies the values on input differently by index.  

I see `maxCharacterCounts={ [9] } `do exactly the same as `maxCharacterCounts={ 9 }`. So I wonder if we can reduce complexity (only one type + removing the Array.flat() call) by using only array format.

Reply:
For me doesn't matter the 3 solutions do same. And they are necessary only because we allowed two types of data in that prop. I still don't see the advantage of allowing a number as acceptable prop type.